ihontaryk/homework_11/mine_sweeper.py

```diff
@@ -0,0 +1,186 @@
+import random
+import argparse
+
+import emoji
+
+CONFIG = {
+    'test': (3, 3, 1),
+    'easy': (10, 10, 10),
+    'medium': (20, 30, 25),
+    'hard': (30, 40, 100)
+}
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('-d', '--difficulty',
+                        action='store',
+                        dest='difficulty',
+                        default='easy',
+                        help='easy | medium | hard')
+
+    return parser.parse_args()
+
+
+class ScreenView:
+
+    def __init__(self, session: "GameSession"):
+        self.session = session
+
+    def update_screen(self, show_mines: bool = False):
+        table = [['--' for _ in range(self.session.width)]
+                 for _ in range(self.session.height)]
+
+        if show_mines:
+            for x, y in self.session.mines:
+                table[y][x] = emoji.emojize(':fire:')
+
+        for x, y in self.session.guesses:
+            table[y][x] = self.session.guesses[x, y]
+
+        format_int = lambda i: format(str(i).zfill(2), "^3")
+
+
+        print("   " + "".join(map(format_int, range(self.session.width))))
+
+        for i, row in enumerate(table):
+            print(format_int(i) + "".join(map(lambda i: format(str(i), "^3"), row)))
+
+    @staticmethod
+    def show_alert(message: str):
+        print(message)
+
+
+class GameSession:
+    NEIGHBOURS = [(0, 1), (0, -1),
+                  (1, 1), (1, 0), (1, -1),
+                  (-1, -1), (-1, 0), (-1, 1)]
+
+    def __init__(self, height, width, mine_count):
+        self.height = height
+        self.width = width
+
+        self.guesses = {}
+        self.mines = []
+
+        self.mines = self.prepare_mines(mine_count)
+        self.screen = ScreenView(self)
+
+    def prepare_mines(self, mine_count: int) -> list:
+        """
+        Prepare list of mines coordinates based on config
+        """
+
+        cells = self.empty_cells()
+        random.shuffle(cells)
+        return cells[:mine_count]
+
+    @staticmethod
+    def receive_user_input() -> str:
+        """
+        Receive coordinates as input from user
+        """
+
+        return input("Make a move in format 'x y': ")
+
+    def normalize_user_input(self, data: str) -> tuple:
+        """
+        Receive input from user, normalize, validate, convert to ints
+        """
+
+        data = tuple(map(int, data.strip().split()))
+        assert len(data) == 2
+        assert data in self.empty_cells() + self.mines
```

Comment:
you do not need empty cells and mines for this check, 
height and width are enough